### PR TITLE
RStudio: 2022.02.3+492 -> 2022.07.1.+554

### PR DESCRIPTION
**Pull request by MrTarantoga**
###### Description of changes
Update RStudio package to current version. It is a small version increase. The change of the build part and the higher integration of quarto needs a change of the patches during build process.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MrTarantoga** [comment]
@ofborg build rstudio

File: pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+-      set(RSTUDIO_DEPENDENCIES_QUARTO_DIR "${RSTUDIO_DEPENDENCIES_DIR}/quarto")
+    endif()
+ 
+ endif()
+@@ -66,15 +64,7 @@
+ 
+ # validate our dependencies exist
+ foreach(VAR RSTUDIO_DEPENDENCIES_DICTIONARIES_DIR
+-            RSTUDIO_DEPENDENCIES_MATHJAX_DIR
+-            RSTUDIO_DEPENDENCIES_PANDOC_DIR
```
Comment (turion) [inline pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch:17]: This is a bit more than just quarto. Just to understand, you're removing this so we can use `nixpkgs` mathjax & pandoc?

File: pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch [hunk 2]
```diff
@@ -0,0 +1,29 @@
+--- a/src/cpp/session/CMakeLists.txt
++++ b/src/cpp/session/CMakeLists.txt
+@@ -45,8 +45,6 @@
+ 
+    if(EXISTS "${RSTUDIO_TOOLS_ROOT}/quarto")
+       set(RSTUDIO_DEPENDENCIES_QUARTO_DIR "${RSTUDIO_TOOLS_ROOT}/quarto")
```
Comment (turion) [inline pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch:6]: So does the "then"-branch always apply? I'm confused what the variable `RSTUDIO_DEPENDENCIES_QUARTO_DIR` is now actually set to, and how the build will find `quarto`.

File: pkgs/applications/editors/rstudio/default.nix [hunk 3]
```diff
@@ -196,7 +197,6 @@ in
       done
 
       rm -r $out/lib/rstudio/{INSTALL,COPYING,NOTICE,README.md,SOURCE,VERSION}
-      rm -r $out/lib/rstudio/bin/{pandoc/pandoc,pandoc}
```
Comment (turion) [inline pkgs/applications/editors/rstudio/default.nix:199]: Why? Do we want to use `pandoc` at runtime?

**Comment by turion** [comment]
Are all the changes necessary for the new version to work? If not, can you separate the version bump (which we can merge quickly) from the other changes?

**Comment by turion** [comment]
> Can we merge it ?

I would not merge before there are some answers to my questions.

**Comment by MrTarantoga** [comment]
@turion I answered you already: https://github.com/NixOS/nixpkgs/pull/185154#discussion_r967069898

**Comment by MrTarantoga** [comment]
And here https://github.com/NixOS/nixpkgs/pull/185154#discussion_r967069898

**Comment by MrTarantoga** [comment]
Well I see, I failed during the usage of the gui function.

**Comment by MrTarantoga** [comment]
grafik

Well, that is what I see. I am a bit suprised with the "pending" label.

**Comment by MrTarantoga** [comment]
Now you should see it? After a bit of googling I understand what I have been done wrong. Well, that is an ... interesting ... workflow for working with comments through the __Files changed__ tab.

**Comment by MrTarantoga** [comment]
I think that is what you meant in your [comment](https://github.com/NixOS/nixpkgs/pull/185154#issuecomment-1246873301)

File: pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch (hunk 1 above)
Comment (turion) [reply pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch:17]: Ok, can you reflect that in the patch filename so noone thinks it's by accident?

**Comment by turion** [comment]
> Now you should see it? After a bit of googling I understand what I have done wrong. Well, that is an ... interesting ... workflow for working with comments through the **Files changed** tab.

Yeah, I had that problem a few times as well when reviewing PRs. Now I can see everything.

File: pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch (hunk 2 above)
Comment (turion) [reply pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch:6]: Ok, but then you could remove the whole `if` construct, right? Maybe that would be clearer.

Comment (MrTarantoga) [reply pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch]: So as you suggested, the pandoc patch and quarto patch is separated. Also the full quarto statement is removed. See commit https://github.com/NixOS/nixpkgs/pull/185154/commits/bb9fcc09880a1f70d0c23f24483b7baa9d8119ac.

File: pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch (hunk 1 above)
Comment (MrTarantoga) [reply pkgs/applications/editors/rstudio/remove-quarto-from-generator.patch]: See answer in https://github.com/NixOS/nixpkgs/pull/185154#discussion_r973501024

**Comment by turion** [comment]
Awesome! Just a tiny thing: Can you rename your second commit to `RStudio: Separate quarto fix and pandoc installation` (typo & package name prefix)

**Comment by turion** [comment]
Thanks a lot!

**Comment by MrTarantoga** [comment]
@turion thank your for your reviews and useful help!

**Comment by alienzj** [comment]
Thank all!

**Pull request merged**